src/java/za/co/mip/ablduck/utilities/CommentParser.java

```diff
@@ -0,0 +1,239 @@
+        commentParseResult.parameterComments = getKeyValueTags(Pattern.compile("@param\\s+(.+?)\\s+(.*)", Pattern.DOTALL));
+    }
+    
+    public void protectInternalComments(){
+        Pattern simpleComment = Pattern.compile("(\\/\\*.*\\*\\/)", Pattern.DOTALL);
+        Matcher m = simpleComment.matcher(comment);
+        Integer commentCount = 0;
+        while (m.find()) {
+            String commentTag = "[comment-" + commentCount.toString() + "]";
+            nestedComments.put(commentTag, m.group(1));
+            comment = comment.replaceFirst(Pattern.quote(m.group(1)), Matcher.quoteReplacement(commentTag));
+            commentCount++;
+        }
+    }
+    
+    public void returnInternalComments(){
+        for (Map.Entry<String, String> c:nestedComments.entrySet()) {
+            comment = comment.replaceFirst(Pattern.quote(c.getKey()), Matcher.quoteReplacement(c.getValue()));
+        }
+    }
+    
+    public HashMap<String, String> getKeyValueTags(Pattern paramRegex) throws IOException{
+        HashMap<String, String> keyvaluetag = new HashMap<String, String>();
+        
+        List<String> tags = javadocParser.parseComment(comment, source);
+        
+        for (int i = 0; i < tags.size(); i++) {
+            Matcher tag = paramRegex.matcher((String)tags.get(i));
+            if(tag.find()) {
+                keyvaluetag.put(tag.group(1), markdown(removeLeadingWhitespace(tag.group(2))));
+              
+                comment = comment.replaceFirst(Pattern.quote((String)tags.get(i)), "");
+            }
+        }
+
+        comment = comment.trim();
+        return keyvaluetag;
+    }
+    
+    public String getValueTag(Pattern paramRegex){
+        String value = "";
+        
+        List<String> tags = javadocParser.parseComment(comment, source);
+        
+        for (int i = 0; i < tags.size(); i++) {
+            Matcher tag = paramRegex.matcher((String)tags.get(i));
+            if(tag.find()) {
+                value = tag.group(1);
+              
+                comment = comment.replaceFirst(Pattern.quote((String)tags.get(i)), "");
+            }
+        }
+
+        comment = comment.trim();
+        return value;
+    }
+    
+    public Boolean getBooleanTag(Pattern paramRegex){
+        Boolean flag = false;
+        
+        List<String> tags = javadocParser.parseComment(comment, source);
+        
+        for (int i = 0; i < tags.size(); i++) {
+            Matcher tag = paramRegex.matcher((String)tags.get(i));
+            if(tag.find()) {
+                flag = true;
+              
+                comment = comment.replaceFirst(Pattern.quote((String)tags.get(i)), "");
+            }
+        }
+
+        comment = comment.trim();
+        return flag;
+    }
+    
+    public String removeLeadingWhitespace(String text){
+        Pattern whitespace = Pattern.compile("\\n\\s+");
+        
+        text = whitespace.matcher(text).replaceAll("\n");

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "text". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1226)